### PR TITLE
Dt 319 retry positioning input field

### PR DESCRIPTION
**Pull request by VilleMiekkoja**

### CODE REVIEW

File: app/component/search/search.scss [hunk 1]
```diff
@@ -23,6 +23,26 @@ $search-form-input-overlay-background-color: #EEF1F3;
   vertical-align: middle;
 }
 
+.search-form {
+  .navigate-or-input-messages {
+    color: $gray-blue;
+    border-bottom-style: dotted;
```
Comment (samuliheljo) [inline app/component/search/search.scss]: We should probably use something smaller style? 
e.g. border-bottom: 1px dotted;
Or what do you think?

File: app/component/search/search.scss [hunk 2]
```diff
@@ -23,6 +23,26 @@ $search-form-input-overlay-background-color: #EEF1F3;
...
+    margin-left: 5px;
+    margin-right: 5px;
+    cursor: pointer;
+  }
+  .navigate-or-input--or {
+    color: $gray;
+  }
+  .navigate-or-input-icon--offline {
+    fill: $gray-blue;
+    font-size: 21px;
```
Comment (samuliheljo) [inline app/component/search/search.scss]: Let's make icons same size

File: app/component/search/autosuggest.cjsx [hunk 3]
```diff
@@ -99,6 +107,7 @@ class Autosuggest extends React.Component
         onSuggestionSelected={@onSuggestionSelected}
         inputAttributes={
           placeholder: @props.placeholder
+          onBlur: @disableInput
           #onBlur: @onSubmit Uh, causes a bug, as it is called after bluring the input after a selection has been done
```
Comment (samuliheljo) [inline app/component/search/autosuggest.cjsx]: Not related to this commit, but what does that "causes a bug" comment mean? should we remove that since we now do not use it?

File: app/component/search/autosuggest.cjsx [hunk 4]
```diff
@@ -60,9 +60,17 @@ class Autosuggest extends React.Component
         {displayText}
       </span>
 
+  componentDidMount: =>
+    if @refs.input.refs.input.value == ""
```
Comment (samuliheljo) [inline app/component/search/autosuggest.cjsx]: You could check this with Hannes and see if he has ideas how to make this more clean. Problem here is that we are touching child component refs directly which is dirty...

File: app/component/search/search-two-fields-container.cjsx [hunk 5]
```diff
@@ -15,6 +16,24 @@ class SearchTwoFieldsContainer extends React.Component
   constructor: ->
     super
 
+    origin = @context.getStore('EndpointStore').getOrigin()
+    dest = @context.getStore('EndpointStore').getDestination()
+
+    @state =
+      if origin.userCurrentPosition
```
Comment (samuliheljo) [inline app/component/search/search-two-fields-container.cjsx]: Is this really "userCurrentPosition" and not "useCurrentPosition"?

Comment (samuliheljo) [inline app/component/search/search-two-fields-container.cjsx]: Could this @state if-else be written in shorter format?

File: app/store/endpoint-store.coffee [hunk 6]
```diff
@@ -74,7 +74,6 @@ class EndpointStore extends Store
       @clearOrigin()
     if @destination.useCurrentPosition
       @clearDestination()
-    @emitChange()
```
Comment (hannesj) [inline app/store/endpoint-store.coffee]: Why was this removed?

File: app/component/search/search-two-fields-container.cjsx [hunk 7]
```diff
@@ -15,6 +16,24 @@ class SearchTwoFieldsContainer extends React.Component
   constructor: ->
     super
 
+    origin = @context.getStore('EndpointStore').getOrigin()
+    dest = @context.getStore('EndpointStore').getDestination()
+
+    @state =
```
Comment (hannesj) [inline app/component/search/search-two-fields-container.cjsx]: why not

```
@state =
  origin:
    userInput: !origin.useCurrentPosition
  dest:
    userInput: !dest.useCurrentPosition
  
```

Also, please use destination rater than dest, as it is used elsewhere in the file

File: app/store/endpoint-store.coffee [hunk 8]
```diff
@@ -25,7 +25,7 @@ class EndpointStore extends Store
         lat: null
         lon: null
         address: null
-    @emitChange()
+    @emitChange('dest')
```
Comment (hannesj) [inline app/store/endpoint-store.coffee]: Change dest to destination for uniformity

File: app/component/search/navigate-or-input.cjsx [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+
+getOriginMessage = () ->
+  <FormattedMessage
+    id="give-origin"
+    defaultMessage='Type origin'
+  />
+
+NavigateOrInput = (props) ->
+  <div className="input-placeholder navigate-or-input">
+      <span onClick={props.setToCurrent}  className="inline-block">
```
Comment (hannesj) [inline app/component/search/navigate-or-input.cjsx]: Inconsistent indentation

File: app/component/search/autosuggest.cjsx (hunk 4 above)
Comment (hannesj) [inline app/component/search/autosuggest.cjsx]: It's ok, just add a comment that it's ugly

File: app/component/search/autosuggest.cjsx (hunk 3 above)
Comment (hannesj) [inline app/component/search/autosuggest.cjsx]: This should be reworded, that we shouldn't do onSubmit on onBlur
